Readme.md

```diff
@@ -1,90 +1,179 @@
 
 ```
-$ ./scripts/compile
+├ main-rule-dir
+│  └examples                                - examples directory
+│     ├ bad                                 - bad examples directory
+│     │  ├ bad-example1
+│     │  │  ├ api-description-document      - API description document that should fail
+│     │  │  └ meta.yaml                     - configuration and metadata for the example
+│     │  └ bad-example2
+│     │     ├ api-description-document
+│     │     └ meta.yaml
+│     └ good
+│        └ good-example1                    - good examples directory
+│           ├ api-description-document      - API description document that should pass
+│           └ meta.yaml                     - configuration and metadata for the example
+├ functions.js                              - validation function definition
+├ rule.yaml                                 - rule configuration and metadata
+└ README.md                                 - generated readme file (do not edit)
 ```
 
-- [Sample Style Guide Readme](./styleguides/sample-styleguide/README.md)
-- [Sample Rule Readme](./rules/sample-rule/README.md)
-- [Sample Good Example Readme](./rules/sample-rule/good-examples/sample-good-example/README.md)
-- [Sample Bad Example Readme](./rules/sample-rule/bad-examples/sample-bad-example/README.md)
+#### 3.3 Write the examples
 
+Write at least one good and bad API description document.
 
+#### 3.4 Write the validation function
 
-### 5. Collaborate, re-use, share, iterate
+Write the validation function to `functions.js`
 
-- Generate the documentation
-- Not all rules can be expressed in current API Description languages, contrinute to their design
-- Fork this repo, add your Rules, Good and Bad Eaxmples, and Styleguides
+Function must be defined by [declaration](https://www.w3schools.com/js/js_function_definition.asp). 
+Function must be exported by 
+    
+```javascript
+module.exports = {
+  myFunction,
+};  
+```
 
-## How it works
+Each function has one input parameter - [minim element](https://github.com/refractproject/minim-api-description) 
+if `minim` is set to `true` in [rule configuration](#write-the-rule-configuration) or (deprecated) JSON object found on desired 
+[path](#supported-targets) in 
+[refract tree](https://github.com/refractproject/refract-spec/blob/master/namespaces/api-description-namespace.md). 
+Function is not executed if no object is found.
+The function must return true if validation passes or a string which describes reason of failure if validation fails.
+You can `require` npm packages (after installing them to `node_modules`) or libraries.
+[lodash](https://lodash.com/) package is available out of the box.
+Any I/O are disallowed for security reasons.
+
+#### 3.5 Write the rule configuration
+
+Write the rule configuration to `rule.yaml`
+
+```md
+title: title of this rule
+intent: intent/description of this rule
+allowedTargets: list of allowed targets
+minim: true
```

Comment:
Do we want to show this to people? Minim by default, right?

Reply:
needs to be there (expl) because i'm not able to determin it otherwise (we need to support non minim functions - there was no flag for that at that time). There is mention in help and readme that it is deprecated